presto-main/src/main/java/com/facebook/presto/cost/CachingCostProvider.java

```diff
@@ -38,22 +38,29 @@
     private final Lookup lookup;
     private final Session session;
     private final TypeProvider types;
+    private final PlanNodeSourceProvider sourceProvider;
 
     private final Map<PlanNode, PlanNodeCostEstimate> cache = new IdentityHashMap<>();
 
-    public CachingCostProvider(CostCalculator costCalculator, StatsProvider statsProvider, Session session, TypeProvider types)
+    public CachingCostProvider(CostCalculator costCalculator, StatsProvider statsProvider, Optional<Memo> memo, Lookup lookup, Session session, TypeProvider types)
     {
-        this(costCalculator, statsProvider, Optional.empty(), noLookup(), session, types);
+        this(costCalculator, statsProvider, memo, lookup, session, types, PlanNode::getSources);
     }
 
-    public CachingCostProvider(CostCalculator costCalculator, StatsProvider statsProvider, Optional<Memo> memo, Lookup lookup, Session session, TypeProvider types)
+    public CachingCostProvider(CostCalculator costCalculator, StatsProvider statsProvider, Session session, TypeProvider types, PlanNodeSourceProvider sourceProvider)
+    {
+        this(costCalculator, statsProvider, Optional.empty(), noLookup(), session, types, sourceProvider);
+    }
+
+    public CachingCostProvider(CostCalculator costCalculator, StatsProvider statsProvider, Optional<Memo> memo, Lookup lookup, Session session, TypeProvider types, PlanNodeSourceProvider sourceProvider)
```

Comment:
Did you check this PR for potential conflicts with https://github.com/prestodb/presto/pull/11267?

Reply:
doesn't overlap i don't think.